onnxruntime/python/tools/transformers/convert_generation.py

```diff
@@ -1706,37 +1766,43 @@ def convert_generation_model(args: argparse.Namespace, generation_type: Generati
                 logger.info(
                     f"{len(initializers)} shared initializers ({[i.name for i in initializers]}) in decoder and init decoder subgraphs are moved to the main graph"
                 )
-            # Update for init decoder subgraph
+
+            # Update init decoder subgraph in preparation to use past present share buffer
             if past_present_share_buffer:
                 logger.info("*****update init decoder subgraph to make past and present share buffer******************")
                 update_decoder_subgraph_past_present_share_buffer(gpt2_init_decoder_model.graph)
+
+            # Update init decoder subgraph in preparation to use DecoderMaskedMultiheadAttention
+            # NOTE: Even if we will not use DecoderMaskedMultiheadAttention in the init decoder subgraph
+            # it makes the runtime changes cleaner if we keep both the init decoder and decoder subgraphs
+            # same in terms of the subgraph inputs.
+            if (
+                args.use_decoder_masked_multihead_attention
+                and not update_decoder_subgraph_use_decoder_masked_multihead_attention(
+                    gpt2_init_decoder_model.graph, is_beamsearch, False
```

Comment:
We update the init decoder subgraph so that they still "consume" the 2 DecoderMaskedMultiheadAttention inputs needed for BeamSearch but they will not use it because the Attention nodes will not get switched. This is needed, otherwise the runtime code becomes ugly to manage if the subgraph inputs for init/non-init decoder become different.